packages/vite/src/node/plugins/resolve.ts

```diff
@@ -222,23 +222,12 @@ export function resolvePlugin(baseOptions: InternalResolveOptions): Plugin {
   }
 }
 
-function tryFsResolve(
-  fsPath: string,
+function tryResolve(
+  file: string,
+  postfix: string,
   options: InternalResolveOptions,
   tryIndex = true
 ): string | undefined {
-  let file = fsPath
-  let postfix = ''
-
-  let postfixIndex = fsPath.indexOf('?')
-  if (postfixIndex < 0) {
-    postfixIndex = fsPath.indexOf('#')
```

Comment:
As https://github.com/vitejs/vite/issues/2346#issuecomment-877701334 said, can we just remove this line so that `#` are always treated as part of file path.